### PR TITLE
[GEOS-9804] The default style of the layergroup is propagated in the layer as default style

### PR DESCRIPTION
**Pull request by taba90**
[GEOS-9804](https://osgeo-org.atlassian.net/browse/GEOS-9804)

The pr adds the possibility to disable the addition of the default layerGroup style in the getCapabilities as this is creating problems to some WMS client eg. QGIS. The default style made also arise problems with INSPIRE compliance as reported in the related Jira ticket. INSPIRE compliance is not being addressed by this pr.
In the WMS settings a new option has been added to enable disable the default style for layergroup with tree modes.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.**

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the `main` branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [x] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

File: doc/en/user/source/services/wms/configuration.rst [hunk 1]
```diff
@@ -64,3 +64,13 @@ The following table shows examples of reasonable values for the request limits:
...
+
+GetCapabilities Settings
+------------------------
+.. list-table::
+   :widths: 20 70
+
+   * - **Option**
+     - **Description**
+   * - Include Default LayerGroup Style In GetCapabilities
+     - Enable/disable the encoding of the default layerGroup style in GetCapabilties responses. By default is set to enabled.
```
Comment (aaime) [inline doc/en/user/source/services/wms/configuration.rst:76]: Does that happen for all types of layer groups? We have some indication that encoding a defatul style for tree based groups (e.g. with visible layers nested inside) are going to cause troubles due to the style being inherited, but not really available for the single layers.

File: src/web/wms/src/main/resources/GeoServerApplication.properties [hunk 2]
```diff
@@ -161,7 +161,8 @@ WMSAdminPage.cacheEntrySize = Max entry size (Kb)
 WMSAdminPage.remoteStyleTime = Remote Styles Time Settings
 WMSAdminPage.remoteStyleTimeout = Timeout in milliseconds
 WMSAdminPage.remoteStyleMaxRequestTime = Max request time in milliseconds
-
+WMSAdminPage.includeDefaultGroupStyleInCapabilities = Include Default LayerGroup Style In GetCapabilities
+WMSAdminPage.getCapabilitiesSettings = GetCapabilities Settings
```
Comment (aaime) [inline src/web/wms/src/main/resources/GeoServerApplication.properties:165]: The WMS page has plenty of capabilities specific settings, if we go for this title, then they should all be grouped toghether. Examples follow.

Root layer and limited SRS list:

image

Scale hint:

image

And well, also the "Service metadata" at the top of the page, it also shows in caps.
Maybe go for "Layer groups capabilities settings" instead?

File: src/web/wms/src/test/java/org/geoserver/wms/web/data/WMSAdminPageTest.java [hunk 3]
```diff
@@ -154,4 +154,14 @@ public void testFeaturesReprojectionDisabled() throws Exception {
         ft.submit("submit");
         assertTrue(wms.isFeaturesReprojectionDisabled());
     }
+
+    @Test
+    public void testIncludeDefaultGroupStyleInCapabilitiesDisabled() throws Exception {
```
Comment (aaime) [inline src/web/wms/src/test/java/org/geoserver/wms/web/data/WMSAdminPageTest.java:159]: Holy cow such a long name... due to the property name (see dedicated comment). We are playing with 80 columns, names should be descriptive, yes, but of reasonable length.

```suggestion
    public void testIncludeDefaultGroupStyleInCapabilitiesDisabled() throws Exception {
```

File: src/wms/src/main/java/org/geoserver/wms/WMSInfo.java [hunk 4]
```diff
@@ -200,4 +200,13 @@ default void setMaxRequestedDimensionValues(int maxRequestedDimensionValues) {
 
     /** Sets the remote style timeout in milliseconds * */
     void setRemoteStyleTimeout(int time);
+
+    default boolean isIncludeDefaultGroupStyleInCapabilities() {
```
Comment (aaime) [inline src/wms/src/main/java/org/geoserver/wms/WMSInfo.java:204]: Too long. Maybe:

```suggestion
    default boolean isDefaultGroupStyleEnabled() {
```

Tried hard to make it shorter than that (still pretty long), came out empty so far.... suggestions welcomed :-D

File: doc/en/user/source/services/wms/configuration.rst (hunk 1 above)
Comment (taba90) [reply doc/en/user/source/services/wms/configuration.rst]: fixed now the option allows to disable the default style for tree types only

File: src/web/wms/src/main/resources/GeoServerApplication.properties (hunk 2 above)
Comment (taba90) [reply src/web/wms/src/main/resources/GeoServerApplication.properties]: fixed

File: src/wms/src/main/java/org/geoserver/wms/WMSInfo.java (hunk 4 above)
Comment (taba90) [reply src/wms/src/main/java/org/geoserver/wms/WMSInfo.java]: fixed

Comment (taba90) [reply src/wms/src/main/java/org/geoserver/wms/WMSInfo.java]: fixed

File: doc/en/user/source/services/wms/configuration.rst (hunk 1 above)
Comment (taba90) [reply doc/en/user/source/services/wms/configuration.rst]: fixed

File: src/web/wms/src/main/resources/GeoServerApplication.properties (hunk 2 above)
Comment (taba90) [reply src/web/wms/src/main/resources/GeoServerApplication.properties]: fixed

File: src/web/wms/src/test/java/org/geoserver/wms/web/data/WMSAdminPageTest.java (hunk 3 above)
Comment (taba90) [reply src/web/wms/src/test/java/org/geoserver/wms/web/data/WMSAdminPageTest.java:159]: fixed

File: src/wms/src/test/java/org/geoserver/wms/capabilities/GetCapabilitiesTransformerTest.java [hunk 5]
```diff
@@ -412,7 +414,169 @@ private void checkVendorSpecificCapsProviders(GetCapabilitiesTransformer tr) thr
...
+                    public boolean isDefaultGroupStyleEnabled() {
+                        return false;
+                    }
+                };
+        GetCapabilitiesTransformer tr2 =
+                new GetCapabilitiesTransformer(wms, baseUrl, mapFormats, legendFormats, null);
+        Document dom2 = WMSTestSupport.transform(req, tr2);
+        String lgStyleTitle =
+                XPATH.evaluate("/WMT_MS_Capabilities/Capability/Layer/Layer/Style/Title", dom2);
+        // the style won't appear
```
Comment (aaime) [inline src/wms/src/test/java/org/geoserver/wms/capabilities/GetCapabilitiesTransformerTest.java:533]: These comments say the style won't appear.. .but it does.

File: src/wms/src/test/java/org/geoserver/wms/capabilities/GetCapabilitiesTransformerTest.java [hunk 6]
```diff
@@ -412,7 +414,169 @@ private void checkVendorSpecificCapsProviders(GetCapabilitiesTransformer tr) thr
...
+        assertEquals(lgStyleTitle, "aLayerGroup style");
+
+        String lgStyleName =
+                XPATH.evaluate("/WMT_MS_Capabilities/Capability/Layer/Layer/Style/Name", dom2);
+        // the style won't appear
+        assertEquals(lgStyleName, "default-style-aLayerGroup");
```
Comment (aaime) [inline src/wms/src/test/java/org/geoserver/wms/capabilities/GetCapabilitiesTransformerTest.java:539]: Same as above.

File: src/wms/src/test/java/org/geoserver/wms/capabilities/GetCapabilitiesTransformerTest.java (hunk 5 above)
Comment (taba90) [reply src/wms/src/test/java/org/geoserver/wms/capabilities/GetCapabilitiesTransformerTest.java]: fixed

File: src/wms/src/test/java/org/geoserver/wms/capabilities/GetCapabilitiesTransformerTest.java (hunk 6 above)
Comment (taba90) [reply src/wms/src/test/java/org/geoserver/wms/capabilities/GetCapabilitiesTransformerTest.java:538]: fixed

**Review by aaime — APPROVED** [review]

**Pull request merged**